opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/Message.java

```diff
@@ -183,11 +193,46 @@ public void serialize(
         public Duration deserialize(final JsonParser parser, final DeserializationContext context)
                 throws IOException {
             try {
+                // TODO Util.decode()
                 return Duration.parse(context.readValue(parser, String.class));
             } catch (DateTimeParseException e) {
                 throw new IOException(e);
             }
         }
     }
 
+    private static class HTMLSerializer extends StdSerializer<String> {
+
+        private static final long serialVersionUID = -2843900664165513923L;
+
+        HTMLSerializer() {
+            this(null);
+        }
+
+        HTMLSerializer(final Class<String> cl) {
+            super(cl);
+        }
+
+        @Override
+        public void serialize(
+                final String string,
+                final JsonGenerator jsonGenerator,
+                final SerializerProvider serializerProvider
+        ) throws IOException {
+            if (string != null) {
+                jsonGenerator.writeString(Util.encode(string));
+            } else {
+                jsonGenerator.writeNull();
+            }
+        }
+    }
+
+    public String toJSON() {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.writeValueAsString(this);
+        } catch (JsonProcessingException e) {
+            return null;
```

Comment:
I'm not sure if this directly would cause the ui to blow on npe when accessing isEmpty or length of the string.

Reply:
However definitely a log statement is necessary